lib/google_mail.rb

```diff
@@ -0,0 +1,65 @@
+require "google/apis/gmail_v1"
+require "googleauth"
+require "json"
+require_relative "env_token_store"
+
+class GoogleMail
+  OOB_URI = "urn:ietf:wg:oauth:2.0:oob".freeze
+  APPLICATION_NAME = "GovWifi Smoke Tests".freeze
+
+  SCOPE = [
+    Google::Apis::GmailV1::AUTH_GMAIL_SEND,
+    Google::Apis::GmailV1::AUTH_GMAIL_MODIFY,
+  ].freeze
+
+  def initialize
+    @service = Google::Apis::GmailV1::GmailService.new
+    @service.client_options.application_name = APPLICATION_NAME
+    @service.authorization = authorize
+  end
+
+  def authorize
+    client_id = Google::Auth::ClientId.from_hash JSON.parse(ENV["GOOGLE_API_CREDENTIALS"])
+    token_store = EnvTokenStore.new ENV
+    authorizer = Google::Auth::UserAuthorizer.new client_id, SCOPE, token_store
+    user_id = "default"
+    credentials = authorizer.get_credentials user_id
+    if credentials.nil?
+      url = authorizer.get_authorization_url base_url: OOB_URI
+      puts "Open the following URL in the browser and enter the " \
+           "resulting code after authorization:\n" + url
+      code = gets
+      credentials = authorizer.get_and_store_credentials_from_code(
+        user_id: user_id,
+        code: code,
+        base_url: OOB_URI,
+      )
+    end
+    credentials
+  end
+
+  def read(query = "")
+    messages = @service.list_user_messages("me", q: query).messages
+    if messages
```

Comment:
Does this repo have a linter enabled? The GOV.UK linter would suggest something like ```return if messages.empty?``` here

Reply:
It does, and it seems fine with the above - happy to change it though.